tests/by-util/test_printf.rs

```diff
@@ -1036,3 +1036,21 @@ fn float_switch_switch_decimal_scientific() {
         .succeeds()
         .stdout_only("1e-05");
 }
+
+#[test]
+fn float_arg_with_whitespace() {
+    new_ucmd!()
+        .args(&["%f", " \u{0020}\u{000d}\t\n0.000001"])
+        .succeeds()
+        .stdout_only("0.000001");
+
+    new_ucmd!()
+        .args(&["%f", "0.1 "])
+        .fails()
+        .stderr_contains("value not completely converted");
+
+    new_ucmd!()
+        .args(&["%f", "\u{2029}0.1"])
+        .fails()
+        .stderr_contains("expected a numeric value");
```

Comment:
Coming from that interesting discovery, do you want to test that printf doesn't actually expand `\t` macro?

i.e. `.args(&["%f", "\\t0.1"])` fails. You'll need to add a comment to explain too, I'm sure others will get confused about this one too.